src/js/apps/patients/worklist/worklist-owned-by_app.js

```diff
@@ -0,0 +1,194 @@
+import _ from 'underscore';
+import moment from 'moment';
+import store from 'store';
+
+import Backbone from 'backbone';
+import Radio from 'backbone.radio';
+
+import { dateSort } from 'js/utils/sorting';
+import intl from 'js/i18n';
+
+import App from 'js/base/app';
+
+import FiltersApp from './filters_app';
+
+import { ListView, LayoutView, SortDropList } from 'js/views/patients/worklist/worklist_views';
+
+const i18n = intl.patients.worklist.worklistApp;
+
+const sortOptions = new Backbone.Collection([
+  {
+    id: 'sortDueAsc',
+    text: i18n.sortOptions.sortDueAsc,
+    comparator(a, b) {
+      if (a.model.get('due_date') === b.model.get('due_date')) {
+        return dateSort(
+          'asc',
+          moment(a.model.get('due_time'), 'HH:mm:ss'),
+          moment(b.model.get('due_time'), 'HH:mm:ss'),
+        );
+      }
+      return dateSort('asc', a.model.get('due_date'), b.model.get('due_date'));
+    },
+  },
+  {
+    id: 'sortDueDesc',
+    text: i18n.sortOptions.sortDueDesc,
+    comparator(a, b) {
+      if (a.model.get('due_date') === b.model.get('due_date')) {
+        return dateSort(
+          'desc',
+          moment(a.model.get('due_time'), 'HH:mm:ss'),
+          moment(b.model.get('due_time'), 'HH:mm:ss'),
+        );
+      }
+      return dateSort('desc', a.model.get('due_date'), b.model.get('due_date'));
+    },
+  },
+  {
+    id: 'sortUpdateAsc',
+    text: i18n.sortOptions.sortUpdateAsc,
+    comparator(a, b) {
+      return dateSort('asc', a.model.get('updated_at'), b.model.get('updated_at'));
+    },
+  },
+  {
+    id: 'sortUpdateDesc',
+    text: i18n.sortOptions.sortUpdateDesc,
+    comparator(a, b) {
+      return dateSort('desc', a.model.get('updated_at'), b.model.get('updated_at'));
+    },
+  },
+]);
+
+export default App.extend({
+  childApps: {
+    filters: FiltersApp,
+  },
+  stateEvents: {
+    'change': 'onChangeState',
+  },
+  onChangeState(state) {
+    store.set(this.stateId, state);
+    if (state.hasChanged('groupId') || state.hasChanged('clinicianId')) {
+      this.restart();
+    }
+  },
+  viewEvents: {
+    'toggle:listType': 'onToggleListType',
+  },
+  initListState() {
+    this.stateId = `worklist-${ this.worklistId }-${ this.worklistType }`;
+
+    const default_state = {
+      sortId: 'sortUpdateDesc',
+      groupId: 'all-groups',
+      clinicianId: this.currentClinician.id,
+    };
+
+    if (!store.get(this.stateId)) {
+      store.set(this.stateId, default_state);
+      this.setState(default_state);
+    } else {
+      this.setState(store.get(this.stateId));
+    }
+  },
+  onBeforeStart({ worklistId, worklistType }) {
+    if (this.isRestarting()) {
+      this.getRegion('list').startPreloader();
```

Comment:
this happens in the `if` and the `else`  let's just make it the first line

Reply:
yeah I was running into an issue with this...lemme see if I can't reproduce it

Reply:
yup, so if I just make `this.getRegion('list')` the first line, it errors because the region is undefined. If keep only the bottom one, I get a "TypeError: pattern is undefined"

Reply:
ah right.. because it's based off the layout.. carry on